packages/gatsby-plugin-google-tagmanager/src/__tests__/gatsby-browser.js

```diff
@@ -110,4 +137,60 @@ describe(`onRouteUpdate`, () => {
 
     expect(window[dataLayerName]).toHaveLength(1)
   })
+
+  it(`sends core web vitals when enabled`, async () => {
+    const { onInitialClientRender } = getAPI(() => {
+      process.env.NODE_ENV = `production`
+    })
+    onInitialClientRender({}, { enableWebVitalsTracking: true })
+
+    // wait 2 ticks to wait for dynamic import to resolve
+    await Promise.resolve()
+    await Promise.resolve()
+
+    jest.runAllTimers()
+
+    expect(window.dataLayer.length).toBe(3)
+    expect(window.dataLayer).toContainEqual({
+      event: `core-web-vitals`,
+      webVitalsMeasurement: {
+        name: `LCP`,
+        id: `1`,
+        value: 300,
+      },
+    })
+    expect(window.dataLayer).toContainEqual({
+      event: `core-web-vitals`,
+      webVitalsMeasurement: {
+        name: `FID`,
+        id: `2`,
+        value: 150,
+      },
+    })
+    expect(window.dataLayer).toContainEqual({
+      event: `core-web-vitals`,
+      webVitalsMeasurement: {
+        name: `CLS`,
+        id: `3`,
+        value: 100,
+      },
+    })
+  })
+
+  it(`sends nothing when web vitals tracking is disabled`, async () => {
+    const { onInitialClientRender } = getAPI(() => {
+      process.env.NODE_ENV = `production`
+    })
+    onInitialClientRender({}, { enableWebVitalsTracking: false })
+
+    jest.runAllTimers()
+
+    // wait 2 ticks to wait for dynamic import to resolve
+    await Promise.resolve()
+    await Promise.resolve()
```

Comment:
Other tests have reverse order of these (not sure if it matters tho)
```suggestion
    // wait 2 ticks to wait for dynamic import to resolve
    await Promise.resolve()
    await Promise.resolve()

    jest.runAllTimers()
```